src/enzo/SetEvolveRefineRegion.C

```diff
@@ -23,68 +27,281 @@ int SetEvolveRefineRegion (FLOAT time)
+	  MustRefineRegionLeftEdge[dim]=EvolveRefineRegionLeftEdge[timestep][dim];
+	  MustRefineRegionRightEdge[dim]=EvolveRefineRegionRightEdge[timestep][dim];
+	}
       }
     }
-  }
-  timestep -= 1;
-  if (timestep < 0) return SUCCESS;
-
-  /* Set RefineRegion to EvolveRefineRegion or innermost StaticRefineRegion */
-
-  staticRegion = 0;
-  while (StaticRefineRegionLevel[staticRegion] != INT_UNDEFINED)
-    staticRegion++;
-  staticRegion -= 1;
-
-  for (i = 0; i < MAX_DIMENSION; i++)
-    if (staticRegion < 0) {
-      RefineRegionLeftEdge[i] = EvolveRefineRegionLeftEdge[timestep][i];
-      RefineRegionRightEdge[i] = EvolveRefineRegionRightEdge[timestep][i];
-    } else {
-      RefineRegionLeftEdge[i] = max(EvolveRefineRegionLeftEdge[timestep][i],
-				    StaticRefineRegionLeftEdge[staticRegion][i]);
-      RefineRegionRightEdge[i] = 
-	min(EvolveRefineRegionRightEdge[timestep][i],
-	    StaticRefineRegionRightEdge[staticRegion][i]);
+
+  } // if((RefineRegionTimeType == 1) || (RefineRegionTimeType == 0) )
+
+
+
+  /* does time evolution for MustRefineRegion ONLY! */
+  if((MustRefineRegionTimeType == 1) || (MustRefineRegionTimeType == 0) ){
+  
+    /* Find closest time step with <time */
+    /* Set time=redshift if that's what we're doing. */
+    if (MustRefineRegionTimeType == 1) {  // redshift
+      time = redshift;
+
+      /* Check to see if the current redshift is within the bounds of the given time file.  This assumes
+	 that EvolveMustRefineRegionTime[0] is the highest redshift and the last output is the lowest (for the 
+	 given track file).  */
+      if(time > EvolveMustRefineRegionTime[0] || time < EvolveMustRefineRegionTime[EvolveMustRefineRegionNtimes-1]){
+	fprintf(stderr,"SetEvolveRefineRegion ERROR for EvolveMustRefineRegions: current simulation redshift is outside of range of track file redshifts!");
+	my_exit(EXIT_FAILURE);
+      }
+
+      
+      for(timestep=0; timestep<EvolveMustRefineRegionNtimes; timestep++){
+	if( time > EvolveMustRefineRegionTime[timestep] ){
+	  break;
+	}
+      }
+    }else{ // code time
+
+      /* Check to see if the current time is within the bounds of the given time file.  This assumes
+	 that EvolveMustRefineRegionTime[0] is the earliest time and the last output is the latest time (for
+	 the given track file). */
+      if(time < EvolveMustRefineRegionTime[0] || time > EvolveMustRefineRegionTime[EvolveMustRefineRegionNtimes-1]){
+	fprintf(stderr,"SetEvolveRefineRegion ERROR for EvolveMustRefineRegion: current simulation time is outside of range of track file redshifts!");
+	my_exit(EXIT_FAILURE);
+      }
+
+      for(timestep=0; timestep<EvolveMustRefineRegionNtimes; timestep++){
+	if( time < EvolveMustRefineRegionTime[timestep] ){
+	  break;
+	}
+      }
     }
+    timestep -= 1;
+    if (timestep < 0) return SUCCESS;
+
+    /* Set MustRefineRegion to EvolveMustRefineRegion */
+
+    for (i = 0; i < MAX_DIMENSION; i++){
+
+      if(timestep==EvolveMustRefineRegionNtimes-1){
```

Comment:
Small spacing change
```suggestion
      if(timestep == EvolveMustRefineRegionNtimes-1){
```